### PR TITLE
Bugfix/manageiq inventory

### PR DESCRIPTION
**Pull request by phospi**
##### SUMMARY
I stumbled across an interesting issue with the inventory script. I noticed that my inventory also contained archived vm's which doesn't make any sense. If vm's are archived on the manageiq/cloudforms side, they are gone. If they are gone, they can never be reached via ansible. So I modified the cloudforms api call to filter for active vm's. And while I was doing that I did some refactoring on the http request as well.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
cloudforms.py

### CODE REVIEW

**Comment by phospi** [comment]
cc @agx @abellotti @cben @dkorn @gtanzillo @yaacov @zgalor

Could someone please review this PR? Thanks.

**Comment by felixfontein** [comment]
I cannot comment on the fix, but you definitely need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

File: changelogs/fragments/720-cloudforms_inventory.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - cloudforms inventory - fixed issue that non-existing (archived) vm's were synced. (https://github.com/ansible-collections/community.general/pull/720).
```
Comment (felixfontein) [inline changelogs/fragments/720-cloudforms_inventory.yml:2]:
```suggestion
  - cloudforms inventory - fixed issue that non-existing (archived) VMs were synced (https://github.com/ansible-collections/community.general/pull/720).
```

**Comment by felixfontein** [comment]
@agx @abellotti @cben @dkorn @gtanzillo @yaacov @zgalor please review this PR.

If nobody reviews it until in ~1.5 weeks, I'll merge the PR so it gets included in the 2.2.0 release.

**Comment by felixfontein** [comment]
Merging now so that #719 needs to be rebased only once.

**Comment by felixfontein** [comment]
@phospi thanks for this contribution!

**Pull request merged**